lib/Sema/TypeCheckType.cpp

```diff
@@ -3402,20 +3415,18 @@ Type TypeChecker::substMemberTypeWithBase(ModuleDecl *module,
           aliasDecl, baseTy,
           aliasDecl->getASTContext());
     }
+  }
 
-    // FIXME: If this is a protocol typealias and we haven't built the
-    // protocol's generic environment yet, do so now, to ensure the
-    // typealias's underlying type has fully resolved dependent
-    // member types.
-    if (auto *protoDecl = dyn_cast<ProtocolDecl>(aliasDecl->getDeclContext())) {
-      ASTContext &ctx = protoDecl->getASTContext();
-      ctx.getLazyResolver()->resolveProtocolEnvironment(protoDecl);
-    }
+  Type memberType;
+  if (aliasDecl) {
+    memberType = aliasDecl->getUnderlyingTypeLoc().getType();
```

Comment:
This duplicates the above check, see my comment about the GSB and getStructuralType()